### ISSUE TITLE
Subpages links doesn't work

### ISSUE THREAD

**Issue by Kiblyn11** [rust-lang/mdBook#1268]
I'm not sure how this should work but I cannot make links works in subpages.

Here my structure:
```
- SUMMARY.md
- folder1
  - README.md
  - folder2
    - README.md
    - stuff.md
  - folder3
    - README.md
    - stuff.md
```

Summary is OK and links works:
```
# Summary

- [folder1](./folder1/README.md)
  - [folder2](./folder1/folder2/README.md)
  - [folder3](./folder1/folder3/README.md)
```

But `folder1/README.md` links doesnt work
```
# folder1

- [folder2](./folder2/README.md)
- [folder3](./folder3/README.md)
```

I get 404 Document not found when clicking on these in the folder1 page.
It tries to access `http://localhost:3000/folder1/folder2/README.html` instead of `http://localhost:3000/folder1/folder2/index.html` (according to the link of the main page)

**Comment by ehuss** [comment]
All files named `README.md` are converted to `index.md`.  The documentation for this is pretty hidden (https://rust-lang.github.io/mdBook/format/config.html?highlight=readme#configuring-preprocessors), it should probably be added to the SUMMARY documentation.  It might also be possible to catch that mistake as a warning, though I'm not sure how hard that would be.

**Comment by Kiblyn11** [comment]
Ok thanks!

**Comment by kssmilik** [comment]
Adding this piece to the `book.toml` config solved this issue to me.

```
[build]
use-default-preprocessors = false

[preprocessor.links]
```

**Comment by xileftenurb** [comment]
It work because you desactivate 
```
[preprocessor.index]
```
who rename all `README.md` file into `index.html`

It seems the index preprocessor change the link in `SUMMARY.md` but not those in the rest of the book.

It look like a bug to me.

**Comment by xileftenurb** [comment]
Also, this is a duplicate of #984

**Comment by xileftenurb** [comment]
The problem is that there is no logic to change the path in file other than SUMMARY.md in the preprocessor.
the processor is defined here and is pretty simple :
https://github.com/rust-lang/mdBook/blob/master/src/preprocess/index.rs

check the path, if it's match "README.md", rename the file to "index.md" and change the path to "index.md"

To rename all link, the preprocessor should open all file (README or not), search for path who point to README, and changed them to  index.md`
I don't know rust well enough to do it myself, sadly.

**Comment by Progdrasil** [comment]
This is blocked by the issue in `pulldown-cmark` https://github.com/raphlinus/pulldown-cmark/issues/423

**Comment by shonfeder** [comment]
Looks like raphlinus/pulldown-cmark#423 has been resolved :tada:  :) 

We'd love to see this fixed. :pray:

**Comment by ehuss** [comment]
Closing as a duplicate of #984. Feel free to leave more comments over there.
